tests/mapobject/test_mapobject.cpp

```diff
@@ -0,0 +1,44 @@
+#include "mapobject.h"
+#include "objecttemplate.h"
+
+#include <QtTest/QtTest>
+
+using namespace Tiled;
+
+class test_MapObject : public QObject
+{
+    Q_OBJECT
+
+private slots:
+    void test_syncWithTemplate();
+};
+
+
+void test_MapObject::test_syncWithTemplate()
+{
+    MapObject obj;
+    Properties props;
+    props["hp"] = 50;
+    obj.addComponent("component1", props);
+    ObjectTemplate templateObj;
+    templateObj.setObject(&obj);
+
+    MapObject mapObject;
+    mapObject.setObjectTemplate(&templateObj);
+
+    mapObject.syncWithTemplate();
+    QCOMPARE(mapObject.componentProperties("component1")["hp"], 50);
+
+    obj.setComponentProperty("component1", "hp", QVariant(55));
```

Comment:
Here we don't need `QVariant(55)` and can just write `55`, because the conversion to `QVariant` happens implicitly.